index.js

```diff
@@ -18,61 +16,29 @@ const pWriteFile = util.promisify(fs.writeFile)
 module.exports = {
   async onPreBuild({ netlifyConfig, packageJson, utils }) {
     const { failBuild } = utils.build
+    const nextConfigPath = await findUp('next.config.js')
 
     validateNextUsage(failBuild)
 
-    if (Object.keys(packageJson).length === 0) {
-      return failBuild(`Could not find a package.json for this project`)
-    }
-
-    const { build } = netlifyConfig
-    const { name, scripts = {}, dependencies = {} } = packageJson
-
-    if (isStaticExportProject({ build, scripts })) {
-      return failBuild(
-        `Static HTML export Next.js projects do not require this plugin. Check your project's build command for 'next export'.`,
-      )
+    if (doesNotNeedPlugin({ netlifyConfig, packageJson, nextConfigPath })) {
+      return
     }
 
-    const hasNextOnNetlifyInstalled = dependencies['next-on-netlify'] !== undefined
-    const hasNextOnNetlifyPostbuildScript =
-      typeof scripts.postbuild === 'string' && scripts.postbuild.includes('next-on-netlify')
-    const isAlreadyUsingNextOnNetlify =
-      (hasNextOnNetlifyInstalled || hasNextOnNetlifyPostbuildScript) && pluginName !== name
-    if (isAlreadyUsingNextOnNetlify) {
-      return failBuild(
-        `This plugin does not support sites that manually use next-on-netlify. Uninstall next-on-netlify as a dependency to resolve.`,
-      )
-    }
-
-    const nextConfigPath = await findUp('next.config.js')
-    if (nextConfigPath !== undefined) {
-      // We cannot load `next` at the top-level because we validate whether the
-      // site is using `next` inside `onPreBuild`.
-      const { PHASE_PRODUCTION_BUILD } = require('next/constants')
-      const { default: loadConfig } = require('next/dist/next-server/server/config')
-
-      // If the next config exists, fail build if target isnt in acceptableTargets
-      const acceptableTargets = ['serverless', 'experimental-serverless-trace']
-      const nextConfig = loadConfig(PHASE_PRODUCTION_BUILD, path.resolve('.'))
-      const isValidTarget = acceptableTargets.includes(nextConfig.target)
-      if (!isValidTarget) {
-        return failBuild(
-          `Your next.config.js must set the "target" property to one of: ${acceptableTargets.join(', ')}`,
-        )
-      }
-    } else {
+    if (nextConfigPath === undefined) {
       // Create the next config file with target set to serverless by default
       const nextConfig = `
           module.exports = {
             target: 'serverless'
           }
         `
       await pWriteFile('next.config.js', nextConfig)
-      console.log(`** Adding next.config.js with target set to 'serverless' **`)
     }
   },
-  async onBuild({ constants: { PUBLISH_DIR, FUNCTIONS_SRC = DEFAULT_FUNCTIONS_SRC } }) {
+  async onBuild({ netlifyConfig, packageJson, constants: { PUBLISH_DIR, FUNCTIONS_SRC = DEFAULT_FUNCTIONS_SRC } }) {
+    if (doesNotNeedPlugin({ netlifyConfig, packageJson })) {
```

Comment:
Should this pass `nextConfigPath` too?
Otherwise, it will not check for the `target` (unlike `onPreBuild()`). So `onPreBuild()` might be skipped, but not `onBuild()`.

Reply:
ah yes you're right!!! thank you :) 

at first i was trying to decide if i should put the `findUp` call inside `doesNotNeedPlugin` but didnt know if i wanted to make that async so i'll just call it again in onBuild i suppose